android/java/org/chromium/chrome/browser/crypto_wallet/activities/NftDetailActivity.java

```diff
@@ -0,0 +1,172 @@
+        tokenStandard.setText(R.string.brave_wallet_nft_erc_721);
+        TextView tokenIdView = findViewById(R.id.token_id_content);
+
+        String tokenStr = String.format(TOKEN_ID_FORMAT, mNftTokenId);
+        SpannableString spannable = new SpannableString(tokenStr);
+
+        if (!TextUtils.isEmpty(mBlockExplorerUrl)) {
+            String url =
+                    String.format(NFT_URL_FORMAT, mBlockExplorerUrl, mContractAddress, mNftTokenId);
+            NoUnderlineClickableSpan linkSpan = new NoUnderlineClickableSpan(this,
+                    R.color.brave_link, (textView) -> { TabUtils.openLinkWithFocus(this, url); });
+            spannable.setSpan(linkSpan, 0, spannable.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        }
+        tokenIdView.setText(spannable);
+        tokenIdView.setMovementMethod(LinkMovementMethod.getInstance());
+        onInitialLayoutInflationComplete();
+    }
+
+    private void setMetadata(PortfolioModel.Erc721MetaData erc721MetaData) {
+        // In case of no errors proceed to assign description and fetch NFT image.
+        if (erc721MetaData.mErrCode == 0) {
+            String description = erc721MetaData.mDescription;
+            String imageUrl = erc721MetaData.mImageUrl;
+            if (!TextUtils.isEmpty(description)) {
+                TextView descriptionContent = findViewById(R.id.description_content);
+                descriptionContent.setText(description);
+            } else {
+                // Hide description layout in case of an empty string.
+                findViewById(R.id.nft_description).setVisibility(View.GONE);
+            }
+            if (!TextUtils.isEmpty(imageUrl)) {
+                loadNftImage(imageUrl);
+            } else {
+                setNftImageAsNotAvailable();
+            }
+        }
+    }
+
+    private void loadNftImage(String imageUrl) {
+        ImageLoader.getLoadNftRequest(imageUrl, this, false)
+                .listener(new RequestListener<Drawable>() {
+                    @Override
+                    public boolean onLoadFailed(GlideException glideException, Object model,
+                            Target<Drawable> target, boolean isFirstResource) {
+                        setNftImageAsNotAvailable();
+                        return false;
+                    }
+
+                    @Override
+                    public boolean onResourceReady(Drawable resource, Object model,
+                            Target<Drawable> target, DataSource dataSource,
+                            boolean isFirstResource) {
+                        target.onResourceReady(
+                                resource, new DrawableCrossFadeTransition(250, true));
+                        return true;
+                    }
+                })
+                .into(mNftImageView);
+    }
+
+    private void setNftImageAsNotAvailable() {
+        mNftImageView.setVisibility(View.GONE);
+        mImageNotAvailableText.setVisibility(View.VISIBLE);
```

Comment:
nit: `AndroidUtils.gone or show`

Reply:
This surely will be convenient for setting the visibility for multiple views at once. 
I don't see the benefit of using it for a single view; I prefer the good old `View#setVisibility()`. 
Let me know if you have a strong preference, and I'll be happy to adapt tough.